bucket/main.tf

```diff
@@ -0,0 +1,44 @@
+###
+### LOCALS
+###
+
+locals {
+  name = "vault-salt-formula"
+}
+
+###
+### DATA SOURCES
+###
+
+data "aws_partition" "current" {}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_region" "current" {}
+
+data "template_file" "bucket_policy" {
+  template = "${file("bucket_policy.json")}"
+
+  vars {
+    bucket_arn = "${aws_s3_bucket.this.arn}"
+  }
+}
+
+resource "aws_s3_bucket" "this" {
+  bucket = "${local.name}"
```

Comment:
Bucket names must be globally unique within an aws partition, so would be good to generate a random identifier to associate to a deployment. Amazon often uses a uuid, which you can generate with a terraform resource, https://www.terraform.io/docs/providers/random/r/uuid.html